motivaa-backend/motivaa-app/src/test/java/backendtesting/RequestMaker.java

```diff
@@ -0,0 +1,66 @@
+package backendtesting;
+
+import io.restassured.RestAssured;
+
+import java.util.Map;
+
+public class RequestMaker {
+
+    public static String initiateGetRequest(String host,
+                                            String endpoint,
+                                            String jSessionId,
+                                            boolean shouldRequestBeLogged) {
+        if (shouldRequestBeLogged) {
+            return RestAssured.given()
+                    .log().all()
+                    .cookie("JSESSIONID", jSessionId)
+                    .post(host+endpoint)
+                    .then()
+                    .statusCode(200)
+                    .log().all()
+                    .extract().response()
+                    .andReturn().asString();
```

Comment:
If I understand it correctly you are converting the json to string then back to json.
If you just return the response you can still use jsonpath later on it:
https://github.com/rest-assured/rest-assured/wiki/Usage#json-using-jsonpath

Reply:
This is an intersting case, I actually convert it to String because I am not using RestAssured's JsonPath library at all. It is using Groovy Gpath as a syntax which I didnt like (before I didn't even know there are multiple JsonPath solutions!)
https://www.javadoc.io/doc/io.rest-assured/json-path/3.0.0/io/restassured/path/json/JsonPath.html

I gave a try to groovy gpath as it could simplify the code, the reason i didnt like is that I didn't find an online parser like https://jsonpath.com/ to test it.

What is your view? I am quite sure groovy gpath is equally capable, it was only a personal preference.

Reply:
Thanks. Decided to keep it like this for now!